eth2util/deposit/deposit_data_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package deposit_test
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"os"
+	"path"
+	"testing"
+
+	eth2p0 "github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/coinbase/kryptology/pkg/signatures/bls/bls_sig"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/eth2util/deposit"
+	"github.com/obolnetwork/charon/tbls/tblsconv"
+	"github.com/obolnetwork/charon/testutil"
+)
+
+var (
+	depositDataFile = path.Join("testdata", "TestDepositData.golden")
+	privKey         = "07e0355752a16fdc473d01676f7f82594991ea830eea928d8e2254dfd98d4beb"
+	withdrawalAddr  = "0xc0404ed740a69d11201f5ed297c5732f562c6e4e"
+	network         = "prater"
+)
+
+func TestDepositData(t *testing.T) {
+	depositData, err := DepositDataFromFile(t, depositDataFile)
```

Comment:
I do not think you need to ever read deposit data from file, just calculate the expected value and do golden file comparison.

Reply:
done!